operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/Controller.java

```diff
@@ -179,26 +187,43 @@ public String successTypeName(UpdateControl<P> result) {
           @Override
           public UpdateControl<P> execute() throws Exception {
             initContextIfNeeded(resource, context);
-            dependents.forEach(dependent -> dependent.reconcile(resource, context));
+            final var result = dependents.entrySet().stream()
+                .map(entry -> {
+                  final var dependent = entry.getValue();
+                  final var name = entry.getKey();
+                  ReconcileResult reconcileResult;
+                  try {
+                    reconcileResult = dependent.reconcile(resource, context);
+                  } catch (Exception e) {
+                    reconcileResult = ReconcileResult.error(resource, e);
```

Comment:
This logic should be inside of the DependetResource (not in the controller), so then it's usable in standalone mode consistenly. But again I'm strongly agains  to indoructe this is this release.

Reply:
To make it usable consistently for all dependent resources, it needs to be in the `Controller` because we don't control the `DependentResource` implementations. We shouldn't think that people will always use one of the classes in the hierarchy, they might implement `DependentResource` directly.

Reply:
There is not way to force correct implementation of an interace in java, but this is already inconsistent this way, since we would want to have in managed and standalone the same behavior so it can be easily migrated (but also just easy to understand) between the two appraoches.

Reply:
But again this is conceptually does not fit in the current model.